camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java

```diff
@@ -116,6 +116,8 @@ public boolean getNext(EtlKey key, BytesWritable payload, BytesWritable pKey) th
         bytes = new byte[origSize];
         buf.get(bytes, buf.position(), origSize);
         pKey.set(bytes, 0, origSize);
```

Comment:
It happened to us in production.

In [Message.scala](https://github.com/apache/kafka/blob/0.8.1/core/src/main/scala/kafka/message/Message.scala#L136), one of the constructors sets `key` to `null`.